### PR TITLE
Add a control policy for Cartesian trajectory publishing

### PR DESCRIPTION
**Pull request by stefanscherzinger**
This adds an additional `CartesianTrajectoryPublisher` as read-only
controller.  It uses and offers the CartesianTrajectoryController's
action interface but does not claim resources nor actuates the robot's
joints.  It just publishes the trajectory's interpolated Cartesian state
as reference PoseStamped and TwistStamped messages.  Its purpose is to
offer an easy way for users to test whether their Cartesian trajectories
_look_ as expected.

### CODE REVIEW

**Comment by stefanscherzinger** [comment]
Documentation is [here](https://github.com/fzi-forschungszentrum-informatik/cartesian_ros_control/tree/add-policy-for-trajectory-visualization/cartesian_trajectory_controller#visualizing-cartesian-trajectories) in the readme.

File: cartesian_trajectory_controller/README.md [hunk 1]
```diff
@@ -120,6 +120,49 @@ jnt_cartesian_traj_controller:
...
+    type: "cartesian_trajectory_publisher/CartesianTrajectoryPublisher"
+    base: "base_link"
+    tip: "tool0"
+    joints:
+       - shoulder_pan_joint
+       - shoulder_lift_joint
+       - elbow_joint
+       - wrist_1_joint
+       - wrist_2_joint
+       - wrist_3_joint
```
Comment (fmauch) [inline cartesian_trajectory_controller/README.md:149]: It's a bit confusing that you add joints though it is not claiming them. Intuitively I would expect, that specifying joints is not necessary.?

**Review by fmauch — COMMENTED** [review]
Though I do see the benefits of having such a tool, this implementation feels a bit "pressed in". We are reusing the trajectory controller to actually only visualize something.

I think, this shows best in the requirement to specify joints for that visualization, where the joint names should not matter.

I do not want to vote against this, completely, but I am hesitating to simply approve this to be merged in.

**Comment by stefanscherzinger** [comment]
> We are reusing the trajectory controller to actually only visualize something.

Yes, that's true. The good thing is that that's easily possible (like in terms of code) with adding a new _read-only_ control policy for this. What I like about this approach is that changes/updates to the cartesian_trajectory_controller's workings will thus always be reflected in the visualizing controller. I would _not_ try to replicate this with a _stand-alone_ Python script, using the cartesian_trajectory_interpolation library.

> Intuitively I would expect, that specifying joints is not necessary.?

It depends. Using the joint states (read-only) allows to start Cartesian trajectories where we are. If we wanted to go without control handles, we would need TF lookups or similar for this feature. I would see this visualizing controller as a duplicate of the original controller that only _visualizes_ the Cartesian motion. I'm personally fine with having to specify the exact same joints for this.

What I see absolutely mandatory (as in helpful) is a quick possibility for users to somehow check the validity of their Cartesian boundary conditions (before filing issues on github) :-)

**Pull request merged**